components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManager.java

```diff
@@ -80,4 +81,62 @@ BrandingPreference replaceBrandingPreference(BrandingPreference brandingPreferen
      */
     void deleteBrandingPreference(String type, String name, String locale)
             throws BrandingPreferenceMgtException;
+
+    /**
+     * This API is used to create a custom text preference.
+     *
+     * @param customText Custom Text preference.
+     * @return the created custom text preference.
+     * @throws BrandingPreferenceMgtException if any error occurred.
+     */
+    CustomText addCustomText(CustomText customText) throws BrandingPreferenceMgtException;
+
+    /**
+     * This API is used to retrieve a custom text preference.
+     *
+     * @param type   Type of the custom text preference.
+     * @param name   Name of the tenant/application where custom text belongs.
+     * @param screen Screen where the custom text needs to be applied.
+     * @param locale Language preference of the custom text.
+     * @return The requested custom text preference.
+     * @throws BrandingPreferenceMgtException if any error occurred.
+     */
+    CustomText getCustomText(String type, String name, String screen, String locale)
+            throws BrandingPreferenceMgtException;
+
+    /**
+     * This API is used to retrieve a resolved custom text preference.
+     *
+     * @param type   Type of the custom text preference.
+     * @param screen Screen where the custom text needs to be applied.
+     * @param name   Name of the tenant/application where custom text belongs.
+     * @param locale Language preference of the custom text.
+     * @return The resolved custom text preference. If not exists return the default custom text preference.
+     * @throws BrandingPreferenceMgtException if any error occurred.
+     */
+    CustomText resolveCustomText(String type, String name, String screen, String locale)
+            throws BrandingPreferenceMgtException;
+
+    /**
+     * This API is used to replace a given custom text preference.
+     *
+     * @param customText Custom Text preference to be added.
+     * @return Updated custom text preference.
+     * @throws BrandingPreferenceMgtException if any error occurred.
+     */
+    CustomText replaceCustomText(CustomText customText)
+            throws BrandingPreferenceMgtException;
+
+    /**
+     * This API is used to delete a custom text preference.
+     *
+     * @param type   Type of the custom text preference.
+     * @param name   Name of the tenant/application where custom text belongs.
+     * @param screen Screen of the custom text.
+     * @param locale Language preference of the custom text.
+     * @throws BrandingPreferenceMgtException if any error occurred.
+     */
+    void deleteCustomText(String type, String name, String screen, String locale)
```

Comment:
make these new methods as default interface methods to preserve backward compatibility for already implemented implementations.

Reply:
Adddressed with https://github.com/wso2-extensions/identity-branding-preference-management/commit/fdd3486cfdc51558c723795cf2fb75b6faafcdc5